### ISSUE TITLE
feature: Allow `meltano init` to create a project in-place

### ISSUE THREAD

**Issue by ReubenFrankel** [meltano/meltano#6806]
### Feature scope

CLI (options, error messages, logging, etc.)

### Current behaviour

Supply a project name interactively:
```
meltano init
```

Supply a project name to the command:
```
meltano init <project-name>
```

### Optimisation
Both of the above approaches require the user to `cd` into the resulting directory that is created for you:

```sh
meltano init project
cd project
```

It would be nice to be able to create a project in-place:

```
meltano init .
```

Here, the project contents should be created as normal in the current working directory. The command output should be adjusted to account for in-place operation (e.g. remove `cd <project-name>` line under the `Next steps` section).

Running within non-empty directories should check for potential file conflicts and preemptively fail the operation, or even detect whether there is an existing Meltano project present (borrow logic from attempting to run a `meltano` command outside of a project - assuming presence of `meltano.yml`?).

### Example use-case
If a user wanted to version control a new project via a remote repository (GitHub or similar), creating it in-place would avoid having to copy/move the contents of the project directory into the cloned empty repository directory.

Current:
```sh
git clone git@github.com:<user>/my-meltano-project.git

meltano init temp-project
cp -a temp-project/. ../my-meltano project

cd my-meltano-project

git add .
git push

# optionally, clean-up
rm -rf temp-project
```

Proposed;
```sh
git clone git@github.com:<user>/my-meltano-project.git
cd my-meltano-project

meltano init .

git add .
git push

# no cleanup necessary
```

### Comparison to `git init`
Initialise a Git repository in a new directory:
![image](https://user-images.githubusercontent.com/60552974/192675002-632e7bc7-d8e1-4fb2-a917-501282240a16.png)

This is the same behaviour as `meltano init`.

Initialise a Git repository in an existing directory:
![image](https://user-images.githubusercontent.com/60552974/192675129-df127aff-152b-4f9b-bbab-34241d9541ac.png)
![image](https://user-images.githubusercontent.com/60552974/192675157-6189ec08-fd11-4a79-9709-8fe76bf6764f.png)

No argument and `.` result in the same behaviour, whereas currently `meltano init` with no arguments initiates interactive setup.

---

Let me know what you think - happy to open a PR as always! :smile:

**Comment by sbalnojan** [comment]
@ReubenFrankel agreed, stumbled over this multiple times. This behaviour makes the onboarding harder, I tend to first create a new repository and then read the init docs. Also this would allow to:

1. Fork a meltano sample repository (e.g. https://github.com/sbalnojan/meltano-example-el/)
2. init it, start working on it right away. 

Right now you have to first init and then copy the contents of the meltano.yml.

**Comment by tayloramurphy** [comment]
@ReubenFrankel love the idea. Would definitely take a PR for it 👍

**Comment by ReubenFrankel** [comment]
Thinking about this a bit more, is this feature request more like 'Allow `meltano init` to create a project in an existing directory'? That's all `meltano init .` would functionally be doing, and I'd expect the same behaviour for any other existing directory.

---

> Running within non-empty directories should check for potential file conflicts and preemptively fail the operation.

Any thoughts on this in particular?
